services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_api.py

```diff
@@ -293,20 +301,14 @@ async def are_all_services_present(
     """
     The dynamic-sidecar stack always expects to have 2 running services
     """
-    async with docker_client() as client:
-        stack_services = await client.services.list(
-            filters={
-                "label": [
-                    f"swarm_stack_name={dynamic_sidecar_settings.SWARM_STACK_NAME}",
-                    f"uuid={node_uuid}",
-                ]
-            }
-        )
-        if len(stack_services) != 2:
-            log.warning("Expected 2 services found %s", stack_services)
-            return False
+    stack_services = await _get_dynamic_sidecar_stack_services(
+        node_uuid, dynamic_sidecar_settings
+    )
+    if len(stack_services) != 2:
+        log.warning("Expected 2 services found %s", stack_services)
```

Comment:
I think we discussed about this. Why a free function whose purpose is to test a condition should log a warning if the condition is e.g. False? Shouldn't be the task of the caller?

Imaging I use this function with a retry like

```python

async def is_service_ready():
   for attempt in Retry(...):
       while attempt:
           if not await are_all_services_present():
               raise TryAgain
   log.warning("Yep, after trying several times services are not present!")  

```
The warnings hidden in the functions are not useful. It is the caller that should determine whether to warn or not the situation.

Reply:
I agree that at least in this case this warning has no additional value. Will drop the message.

It wasn't an option for the caller to log the same message as above, it would make the code more complicated.